status/health/window/key_error_source.go

```diff
@@ -243,8 +333,8 @@ func (m *multiKeyHealthyIfNotAllErrorsSource) HealthStatus(ctx context.Context)
 	}
 }
 
-func (m *multiKeyHealthyIfNotAllErrorsSource) pruneOldKeys(store TimedKeyStore, maxAge time.Duration) {
-	curTime := m.timeProvider.Now()
+func pruneOldKeys(store TimedKeyStore, maxAge time.Duration, timeProvider TimeProvider) {
```

Comment:
Made this a function so it can be reused between the 2, I don't feel strongly though

Reply:
Personally I feel this should be a method of the `TimedKeyStore`.

Reply:
Good call, done